### PR TITLE
fix(react-chart): arg and val scale inconsistency

### PR DESCRIPTION
**Pull request by DmitryBogomolov**
Fixes https://github.com/DevExpress/devextreme-reactive/issues/2078.

Removes `fixOffset` utils - moves it into `scaleBand` wrapper.

### CODE REVIEW

File: packages/dx-chart-core/src/utils/scale.ts [hunk 1]
```diff
@@ -11,9 +11,12 @@ import {
...
 export const scaleLinear: FactoryFn = d3ScaleLinear as any;
 /** @internal */
-export const scaleBand: FactoryFn = () => (
-  d3ScaleBand().paddingInner(0.3).paddingOuter(0.15) as any
-);
+export const scaleBand: FactoryFn = () => {
+  const scale = d3ScaleBand().paddingInner(0.3).paddingOuter(0.15);
+  const wrapper = (value: any) => scale(value)! + scale.bandwidth() / 2;
+  Object.assign(wrapper, scale);
+  return wrapper as ScaleObject;
```
Comment (MaximKudriavtsev) [inline packages/dx-chart-core/src/utils/scale.ts]: Why we don't use new syntax?

```js
return { ...wrapper, ...scale }
```

File: packages/dx-chart-core/src/utils/scale.ts [hunk 2]
```diff
@@ -246,7 +231,24 @@ const invertBandScaleBounds = (scale: ScaleObject, range: NumberArray): DomainBo
   ];
 };
 
+// tslint:disable-next-line: ban-types
+const makeScaleHelper = <T extends Function>(choices: [T, T]) => {
+  const func: any = (scale: ScaleObject, ...args: any[]) => {
+    const choosen = choices[Number(!!scale.bandwidth)];
```
Comment (MaximKudriavtsev) [inline packages/dx-chart-core/src/utils/scale.ts]: Maybe we should write the next sentence more cleary? I mean something like this
```js
const choosen = scale.bandwidth === 0 ? choices[0] : choices[1]
```

File: packages/dx-chart-core/src/utils/scale.ts (hunk 1 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/utils/scale.ts]: That's not the same - `assign` copies all keys from `scale` to `wrapper`, spread copies all keys from `wrapper` and `scale` to a new object.

Comment (MaximKudriavtsev) [reply packages/dx-chart-core/src/utils/scale.ts]: But a function result is same

File: packages/dx-chart-core/src/utils/scale.ts (hunk 2 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/utils/scale.ts]: `scale.bandwidth` is a function (if exists) - not a number. That would be

```js
const choosen = 'bandwidth' in scale ? choices[0] : choices[1]
```

The question is - will accessing `choices` variable twice be more clear?

Comment (MaximKudriavtsev) [reply packages/dx-chart-core/src/utils/scale.ts]: I think so.

@Krijovnick what do you think about this?

File: packages/dx-chart-core/src/utils/scale.ts (hunk 1 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/utils/scale.ts]: How can it be the same?

Comment (MaximKudriavtsev) [reply packages/dx-chart-core/src/utils/scale.ts]: My mistake, I didn't see that wrapper is a function.

File: packages/dx-chart-core/src/utils/scale.ts (hunk 2 above)
Comment (DmitryBogomolov) [reply packages/dx-chart-core/src/utils/scale.ts]: That code is no more.

Comment (Krijovnick) [reply packages/dx-chart-core/src/utils/scale.ts]: The last variant is more clear.

**Pull request merged**